src/test/java/net/minecraftforge/debug/misc/RegistryCodecTest.java

```diff
@@ -0,0 +1,89 @@
+ */
+
+package net.minecraftforge.debug.misc;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.mojang.datafixers.util.Pair;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.DataResult;
+import com.mojang.serialization.JsonOps;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.nbt.NbtOps;
+import net.minecraft.nbt.Tag;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.level.block.Blocks;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent;
+import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
+import net.minecraftforge.registries.ForgeRegistries;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+@Mod("registry_codec_test")
+public class RegistryCodecTest
+{
+
+    private static final Logger LOGGER = LogManager.getLogger("Codec Registry Test");
+
+    /**
+     *     This Codec can serialize and deserialize a Pair<Item, Block>
+     *     The result Json (or nbt equivalent) will have this structure :
+     *     {
+     *         "block": "block_registry_name",
+     *         "item": "item_registry_name"
+     *     }
+     */
+    private static final Codec<Pair<Block, Item>> CODEC = RecordCodecBuilder.create(codecInstance -> codecInstance.group(
+            ForgeRegistries.BLOCKS.getCodec().fieldOf("block").forGetter(Pair::getFirst),
+            ForgeRegistries.ITEMS.getCodec().fieldOf("item").forGetter(Pair::getSecond)
+    ).apply(codecInstance, Pair::of));
+
+    public RegistryCodecTest()
+    {
+        FMLJavaModLoadingContext.get().getModEventBus().addListener(this::commonSetup);
+    }
+
+    public void commonSetup(final FMLCommonSetupEvent event)
+    {
+        //Create our Json to decode
+        JsonObject json = new JsonObject();
+        json.addProperty("block", "minecraft:diamond_block");
+        json.addProperty("item", "minecraft:diamond_pickaxe");
+
+        //Decode our Json and log an info in case of success or a warning in case of error
+        DataResult<Pair<Pair<Block, Item>, JsonElement>> result = CODEC.decode(JsonOps.INSTANCE, json);
+        result.resultOrPartial(LOGGER::warn).ifPresent(pair -> LOGGER.info("Successfully decoded a diamond block and a diamond pickaxe from json to Block/Item"));
+
+        //Create a Pair<Block, Item> to test the serialization of our codec
+        Pair<Block, Item> pair = Pair.of(Blocks.DIAMOND_BLOCK, Items.DIAMOND_PICKAXE);
+
+        //Serialize the Pair to NBT, and log an info in case of success or a warning in case of error
+        DataResult<Tag> result2 = CODEC.encodeStart(NbtOps.INSTANCE, pair);
+        result2.resultOrPartial(LOGGER::warn).ifPresent(tag -> LOGGER.info("Successfully encoded a Pair<Block, Item> to a nbt tag: " + tag));
```

Comment:
Use logger argument substitution instead of manual string concatenation.
```suggestion
        result2.resultOrPartial(LOGGER::warn).ifPresent(tag -> LOGGER.info("Successfully encoded a Pair<Block, Item> to a nbt tag: {}", tag));
```